### DISCUSSION TITLE
BTP Message Exchange between chains other than ICON

### DISCUSSION THREAD

**Discussion by manishbista28** [icon-project/icon-bridge#655] [General]
We currently have relay chain APIs for ICON, BSC, NEAR, HMNY and SNOW.
Though we have done testnet deployments where ICON is paired with HMNY, BSC and NEAR, we have not tried BTP Message Exchange between these chain themselves yet e.g. between BSC and NEAR. 

Discussion topics:
1. what are the changes that need to be done on existing source code (e.g. deployment scripts & tests)
2. what does it mean for the contracts already deployed separately
3. is the task likely to affect (or block) other concurrent tasks (e.g. other chain and app integrations)
4. ...

Github Issues:
https://github.com/icon-project/icon-bridge/issues/689
https://github.com/icon-project/icon-bridge/issues/733

**Comment by andrii-kl** [comment]
It would be great to define our deployment strategy.

Are we planning to run 1 Relay for 1 chain pairs or 1 Relay for all of them?

It brings special features to the deployment process, that is not ready  to work with more than two chains now.

**Comment by manishbista28** [comment]
As a first step, we have started setting up an environment with multiple chains attached. 
Progress of the task is noted on case #689 

Things learned during this task will help define deployment scripts and tests to automate the job.

**Comment by manishbista28** [comment]
After confirming multi-chain transfer and the extent of changes needed through #689 , the change needed to deployment script will be worked from #733

**Comment by manishbista28** [comment]
We've updated deployment script to integrate new chains and ran some manual tests.
During this process, we've identified that we need to thoroughly examine and document BMC contract primarily to understand the multi-chain transfer methods (routing and such).
